core/src/Kokkos_Complex.hpp

```diff
@@ -64,10 +64,10 @@ class
 
   //! Copy constructor.
   KOKKOS_DEFAULTED_FUNCTION
-  complex(const complex&) noexcept = default;
+  constexpr complex(const complex&) noexcept = default;
```

Comment:
do we even need to make it `constexpr`? thinking of https://stackoverflow.com/questions/20810378/should-i-mark-a-compiler-generated-constructor-as-constexpr

Reply:
Indeed, I went a bit fast on those. Good catch!